cpp/tests/strings/replace_regex_tests.cpp

```diff
@@ -162,6 +162,17 @@ TEST_F(StringsReplaceRegexTest, WordBoundary)
   CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(*results, expected);
 }
 
+TEST_F(StringsReplaceRegexTest, Alternation)
+{
+  cudf::test::strings_column_wrapper input(
+    {"16  6  brr  232323  1  hello  90", "123 ABC 00 2022", "abé123  4567  89xyz"});
+  auto results = cudf::strings::replace_re(
+    cudf::strings_column_view(input), "(^|\\s)\\d+(\\s|$)", cudf::string_scalar("_"));
```

Comment:
Just realized that this is testing for `(^|\\s)` only. Can you add another test with `(\\s|^)` too?

Reply:
Yes, that is a good idea.